infrastructure/tasks/15-functions_sync.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Run this task to sync Azure Functions staging slot
+ * to the source control repository code branch
+ * specified in the configuration file (`tfvars.json`).
+ * 
+ * This is equivalent to push the "Sync" button in the
+ * Azure portal -> Functions -> Deployments blade.
+ * 
+ * This task assumes that the following resources are already created:
+ *  - Resource group
+ *  - Functions (app service)
+ * 
+ */
+// tslint:disable:no-console
+// tslint:disable:no-any
+
+import { login } from "../../lib/login";
+
+import readConfig from "../../lib/config";
+const config = readConfig(__dirname + "/../tfvars.json");
+
+import webSiteManagementClient = require("azure-arm-website");
+
+export const run = async () => {
+  if (!config.functionapp_git_repo) {
+    return Promise.resolve();
```

Comment:
perhaps an error can be thrown here?

Reply:
by looking at the code below, if you reply with a reject, the error will be printed on console

Reply:
actually the git integration is considered optional (this parameter can be empty in the configuration file). should we switch the integration to mandatory ?

moreover the task that creates the functions atm creates a staging functions slot, which is currently unused as we deploy directly to the production slot. should we change the task to avoid creating this further slot ?

Reply:
at least print a message saying that git integration is not configured

re slots: let's keep it as is, there's a story about consolidating the envs